dataactvalidator/filestreaming/loadFile.py

```diff
@@ -1,53 +1,216 @@
 import os
+import pandas as pd
+import boto
+import glob
 from dataactvalidator.filestreaming.loaderUtils import LoaderUtils
 from dataactvalidator.interfaces.validatorValidationInterface import ValidatorValidationInterface
-from dataactvalidator.filestreaming.csvS3Reader import CsvS3Reader
 from dataactcore.models.domainModels import CGAC,ObjectClass,ProgramActivity,SF133
 from dataactcore.config import CONFIG_BROKER
+from dataactvalidator.filestreaming.csvS3Reader import CsvS3Reader
+from sqlalchemy import and_
+
 
 def loadCgac(filename):
-    LoaderUtils.loadCsv(filename,CGAC,ValidatorValidationInterface(),{"cgac":"cgac_code","agency":"agency_name"},{"cgac_code":{"pad_to_length":3,"skip_duplicates":True}})
+    interface = ValidatorValidationInterface()
+    model = CGAC
+
+    # for CGAC, delete and replace values
+    interface.session.query(model).delete()
+    interface.session.commit()
+
+    # read CGAC values from csv
+    data = pd.read_csv(filename, dtype=str)
+    # toss out rows with missing CGAC codes
+    data = data[data['CGAC'].notnull()]
```

Comment:
As it turns out, I added this to handle a missing CGAC code value in our .csv. That missing value should actually be `000`, so ended up removing this line.